core/src/main/java/org/elasticsearch/index/query/SpanOrQueryBuilder.java

```diff
@@ -53,6 +70,55 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     }
 
     @Override
+    protected Query doToQuery(QueryParseContext parseContext) throws IOException {
+        SpanQuery[] spanQueries = new SpanQuery[clauses.size()];
+        for (int i = 0; i < clauses.size(); i++) {
+            Query query = clauses.get(i).toQuery(parseContext);
+            assert query instanceof SpanQuery;
```

Comment:
we might have the same problem with LateParsingQuery here too right? then this should become an if instead? I guess we might have the same somewhere else too?